src/utils/math_utils.py

```diff
@@ -1,19 +1,15 @@
 import numpy as np
+from numba import jit
 from scipy.spatial import ConvexHull
 
 
-def _unit_vector(vector):
-    """ Returns the unit vector of the vector. """
-    return vector / np.linalg.norm(vector)
-
-
-def angle_between(v1, v2):
-    """ Returns the angle in degrees between vectors 'v1' and 'v2'. """
-    v1_u = _unit_vector(v1)
-    v2_u = _unit_vector(v2)
-    dot_product = np.dot(v1_u, v2_u)
-    rad = np.arccos(np.clip(dot_product, -1.0, 1.0))
-    return np.degrees(rad)
+@jit(nopython=True)
+def vector_angle(u, v=np.array([0., 0., 1.])):
```

Comment:
Dit is zeker een stuk sneller. Oude code wss wel handig al willen we de code volledig parallel maken.

Reply:
Dit is nog steeds single-threaded (voor zover numpy single-threaded is).